python/lsst/pex/exceptions/exceptions.cc

```diff
@@ -0,0 +1,155 @@
+// module, preparing the arguments, and calling that function, along with the very verbose error
+// handling required by the Python C API.
+//
+// If any point we fail to translate the exception, we print a Python warning and raise the built-in
+// Python RuntimeError exception with the same message as the C++ exception.
+static void raiseLsstException(py::object &pyex) {
+    PyObject* module = PyImport_ImportModule("lsst.pex.exceptions.wrappers");
+    if (!module) {
+        tryLsstExceptionWarn("Failed to import C++ Exception wrapper module.");
+    } else {
+        PyObject* translate = PyObject_GetAttrString(module, "translate");
+        if (!translate) {
+            tryLsstExceptionWarn("Failed to find translation function for C++ Exceptions.");
+        } else {
+            // Calling the Python translate() returns an instance of the appropriate Python
+            // exception that wraps the C++ exception instance that we give it.
+            PyObject* instance = PyObject_CallFunctionObjArgs(translate, pyex.ptr(), NULL);
+            if (!instance) {
+                // we actually expect a null return here, as translate() should raise an exception
+                tryLsstExceptionWarn("Failed to translate C++ Exception to Python.");
+            } else {
+                PyObject* type = PyObject_Type(instance);
+                PyErr_SetObject(type, instance);
+                Py_DECREF(type);
+                Py_DECREF(instance);
+            }
+            Py_DECREF(translate);
+        }
+        Py_DECREF(module);
+    }
+}
+
+PYBIND11_PLUGIN(_exceptions) {
+    py::module mod("_exceptions", "Access to the classes from the pex_exceptions library");
+
+    py::class_<Tracepoint> clsTracepoint(mod, "Tracepoint");
+
+    clsTracepoint.def(py::init<char const*, int, char const*, std::string const &>())
+        .def_readwrite("_file", &Tracepoint::_file)
+        .def_readwrite("_line", &Tracepoint::_line)
+        .def_readwrite("_func", &Tracepoint::_func)
+        .def_readwrite("_message", &Tracepoint::_message);
+
+    py::class_<Exception> clsException(mod, "Exception");
+
+    clsException.def(py::init<char const*, int, char const*, std::string const &>())
+        .def(py::init<std::string const &>())
+        .def("addMessage", &Exception::addMessage)
+        .def("getTraceback", &Exception::getTraceback)
+        .def("addToStream", &Exception::addToStream)
+        .def("what", &Exception::what)
+        .def("getType", &Exception::getType)
+        .def("clone", &Exception::clone)
+        .def("asString", [](Exception &e){
+            std::ostringstream stream;
+            e.addToStream(stream);
+            return stream.str();
+        })
+        .def("__repr__", [](Exception &e){std::stringstream s; s<<"Exception('"<<e.what()<<"')"; return s.str();});
+
+    py::class_<IoError> clsIoError(mod, "IoError", py::base<Exception>()); 
```

Comment:
Is there a reason why `Exception` used as a base in this declaration instead of true base `RuntimeError`? If yes then a general comment would be useful.